src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -771,9 +772,31 @@ private async Task EnsureInitializeAsync()
             try
             {
                 // If already initialized, need not be on the UI thread
-                if (!_initialized)
+                if (_initialized)
                 {
-                    _initialized = true;
+                    // Check if the cache is initialized.
+                    // It is possible that the cache is not initialized, since,
+                    // the solution was not saved and/or there were no projects in the solution
+                    if (!_cacheInitialized && _solutionOpenedRaised)
+                    {
+                        await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                        await EnsureNuGetAndVsProjectAdapterCacheAsync();
+                    }
+
+                    return;
+                }
+
+                try
+                {
+                    // Ensure all initialization finished when needed, it still runs as async but only prevents _initialized set true too early.
+                    // Getting null value in GetVSSolutionProperty randomly for _vSolution made this change.
+                    await SemaphoreLock.WaitAsync();
```

Comment:
since there's only one main thread, and this code switches to it 5 lines down, we can use that to avoid needing to do thread synchronisation and therefore avoid needing a lock.

Reply:
Ok. I'll try but for now I'm removing this changes for another follow up [issue#10352](https://github.com/NuGet/Home/issues/10352)